src/components/FilterBar.test.jsx

```diff
@@ -0,0 +1,47 @@
+import { render, screen } from '@testing-library/react';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import FilterBar from './FilterBar';
+
+const renderWithTheme = (ui) =>
+  render(<ThemeProvider theme={createTheme()}>{ui}</ThemeProvider>);
+
+const setQueryParam = () => {};
+
+describe('FilterBar', () => {
+  test('SortBy filter (deadline.asc) rendered', () => {
+    const params = { sortBy: 'deadline.asc' };
+    renderWithTheme(<FilterBar queryParams={params} {...{ setQueryParam }} />);
+    expect(
+      screen.getByText('Deadline (Earliest to Latest)')
+    ).toBeInTheDocument();
+  });
+  test('SortBy filter (deadline.desc) rendered', () => {
+    const params = { sortBy: 'deadline.desc' };
+    renderWithTheme(<FilterBar queryParams={params} {...{ setQueryParam }} />);
+    expect(
+      screen.getByText('Deadline (Latest to Earliest)')
+    ).toBeInTheDocument();
+  });
+  test('SortBy filter (amount.desc) rendered', () => {
+    const params = { sortBy: 'amount.desc' };
+    renderWithTheme(<FilterBar queryParams={params} {...{ setQueryParam }} />);
+    expect(screen.getByText('Amount (High to Low)')).toBeInTheDocument();
+  });
+  test('SortBy filter (amount.asc) rendered', () => {
```

Comment:
These tests seem to be testing `FilterDropdown` more than `FilterBar`. I think what you want to do is have tests that check:
- How the component renders by default. E.g. `"renders amount filter and sorting dropdown by default"`
- How the component renders when params are given (you could do this as one test or as a single test per param). E.g. `"renders passed in sorting option"`.
- How the component renders when experiments are enabled (again, either as one test or as a single test per experiment). E.g. `"renders majors filter when experiment enabled"`.
- How the component interacts with `setQueryParam`, maybe once per filter value. This involves interacting with the UI element itself using `get*By*` and setting some property. See [this guide](https://reactjs.org/docs/test-utils.html#act) for an example. E.g. `"calls setQueryParam when sorting option changed"`

That being said, feel free to merge these changes in separate stages/PRs. Some of our test files are actually missing some of these pieces too.

Reply:
You can even get some of the bullets done above and then file a separate issue to add the remaining tests.